backend/business-partner-agent/src/main/java/org/hyperledger/bpa/impl/aries/ProofManager.java

```diff
@@ -136,6 +160,95 @@ public void sendPresentProofRequest(@NonNull UUID partnerId, @NonNull RequestPro
         }
     }
 
+    public void rejectPresentProofRequest(UUID BPAPresentationExchangeId, String explainString) {
+        peRepo.findById(BPAPresentationExchangeId).ifPresentOrElse(pe -> {
+            try {
+                sendPresentProofProblemReport(pe.getPresentationExchangeId(), explainString);
+                // after sending rejection notice, delete aries copy.
+                ac.presentProofRecordsRemove(pe.getPresentationExchangeId());
+                // after sending removing aries copy, remove BPA copy
+
+            } catch (IOException e) {
+                log.error("aca-py not reachable.", e);
+                return;
+            } catch (AriesException e) {
+                log.error("aca-py wallet item not found, attempting BPA delete");
+            }
+            // TODO preserve rejected presentations.
+            // rejected is currently not a valid state in the aries present_proof_v1.0
+            // protocol
+            peRepo.delete(pe);
+        }, () -> {
+            log.error("BPA has no record of presentationexchange with id: {]", BPAPresentationExchangeId);
+        });
+
+    }
+
+    public void presentProof(UUID BPAPresentationExchangeId) {
+        peRepo.findById(BPAPresentationExchangeId).ifPresentOrElse(pe -> {
+            try {
+                ac.presentProofRecordsGetById(pe.getPresentationExchangeId()).ifPresentOrElse(
+                        (record) -> {
+                            presentProof(record);
+                        },
+                        () -> {
+                            log.error("PresentationExchangeRecord not found with Id: {}",
+                                    pe.getPresentationExchangeId());
+                        });
+            } catch (IOException e) {
+                log.error("aca-py not reachable.", e);
+                return;
+            }
+            peRepo.updateState(pe.getId(), PresentationExchangeState.PRESENTATIONS_SENT);
```

Comment:
shouldn't this  taken care of in the event handler? ah i see. thats because it's related to my question above, it's handled in the other handler. do we need to track this twice?